Add Wallet Recovery + Page Updates (#66)
lukso-network/website-support-center: docs/extension/guides/set-wallet-recovery.md

---

> sidebar_label: '🦊 Set Wallet Recovery'

Reviewer: I would not use the fox emoji as it implies MetaMask but the feature is not MetaMask related. Any wallet can work.

sidebar_position: 6

> description: 'Extension Support: How to set up wallet recovery'

Reviewer: This text will be avail. here on search engines:

<img width="612" alt="image" src="https://github.com/lukso-network/website-support-center/assets/477945/8e2f5732-1118-48d9-8c07-9002afdef18d">

So humans will read it.

```suggestion
description: 'How to set up wallet recovery on the LUKSO Universal Profile Extension?'
```

or something like this

---

# Set Wallet Recovery

> If you want an autonomous backup for your Universal Profile, you can add an existing wallet as a recovery controller.

Reviewer: cf other comment "autonomous backup" seems weird to me.

I would change "existing wallet" with "another wallet"

This wallet may not exist yet. The main idea is to set another wallet.

## Preparation

To add a new controller account to your Universal Profile, you will need the [Universal Profile Browser Extension](https://chromewebstore.google.com/detail/universal-profiles/abpickdkkbnbcoepogfhkhennhfhehfn) as well as an [Ethereum wallet with custom network support](../../general/wallet-support.md) installed **in the same [supported browser](../introduction.md#which-browsers-does-the-extension-support)**.

:::tip Wallet Support

> The [Wallet Recovery Setup](https://my.universalprofile.cloud/3rd-party/add-recovery) officially supports **MetaMask**as the wallet. If you experience issues with other wallets, please try again using MetaMask.

Reviewer: ```suggestion
The [Wallet Recovery Setup](https://my.universalprofile.cloud/3rd-party/add-recovery) supports **MetaMask** as the wallet. If you experience issues with other wallets, please try again using MetaMask.
```

> The [Wallet Recovery Setup](https://my.universalprofile.cloud/3rd-party/add-recovery) supports **MetaMask** as the wallet. If you experience issues with other wallets, please try again using MetaMask.

Reviewer 2: We'll also "promote" Ledger and Coinbase wallet. Maybe later we'll add more logos

:::

> After installing both extensions in your browser, you can set up the recovery account. Based on the wallet, you can either:

Reviewer: im not sure about the grammar of "Based on the wallet"

maybe something like "depending on your wallet"

but im not 100% sure and not english native. Just sharing a though here

- use one of the _default wallet accounts_
- import an _account from a private key_
- connect an _account from your hardware wallet_

Open the wallet and add or select the account from a source of your choice:

<div style={{textAlign: 'center'}}>

<img

>     src="/img/extension/3rd-party-import.png"

Reviewer 2: We changed a bit the screens, so let's take new screenshot as soon as we merge to develop (I'll ping you in Discord 👍

    alt="3rd-party Account Import"
    width="600"
/>

</div>

## Connect the Extensions

> After your recovery account is set in the wallet, head to the [Wallet Recovery Setup](https://my.universalprofile.cloud/3rd-party/add-recovery) and connect both extensions to the service.

Reviewer: ```suggestion
After your recovery account is set in the wallet, head to the [Wallet Recovery Setup](https://my.universalprofile.cloud/3rd-party/add-recovery) and connect both extensions.
```

![Connect the Extensions](/img/extension/set-recovery-1.png)

## Execute the Controller Transaction

The address of the connected wallet account is then used to execute the controller transaction from the Universal Profile Extension.

![Execute the Controller Transaction](/img/extension/set-recovery-2.png)

After the transaction has been executed, your new recovery controller will be set and will show up in the _Controllers_ menu of the Universal Profile.

:::tip Set Controller Names

> By default, the new recovery controller will be named after the account's address. To better distinguish between different controllers, it's recommended to _rename them locally_ by opening the controller view and clicking the pencil next to its name.

Reviewer 2: Ok so I asked Extension team to do the following: if we add a recovery controller key to the profile, we automatically name it "Recovery key?". So it's simpler for the user to identify this controller, and then he can rename it manually. This will be available when we release publicly
